### ISSUE TITLE
Validate is not rerun when props change

### ISSUE THREAD

**Issue by Aaron7** [erikras/redux-form#1965]
We use props inside our validators. However, `validate` is only called if the `values` have updated, not the `props`.

For example, with:
```javascript
const validate = (values, props) => {
  const errors = {}
  if (props.exampleProperty && !values.exampleField) {
    errors.exampleField = 'This field is required.'
  }
  return errors
}
```
If `props.exampleProperty` changes from `false` to `true` and `values` has not changed, the validator is not rerun.

-

One possible solution would be to add a config called `shouldValidate` which would allow you to implement a function which returns `true` if `validate` should be called.

I'm happy to make any changes. Thanks.

**Comment by oakley808** [comment]
When I enter valid text into a form field it validates as good. When I then change it to something invalid, the validation function doesn't seem to rerun, and my form can be submitted with bad entry data.

**Comment by Aaron7** [comment]
With the addition of the `shouldValidate` config property added in #1995, it is now possible to revalidate on props change.

**Comment by erikras** [comment]
Fix published in [`v6.2.0`](https://github.com/erikras/redux-form/releases/tag/v6.2.0).
